### PR TITLE
Clean up - documentation -  reference contrib section convert to markdown

### PR DESCRIPTION
**Pull request by lb-**
-  Part of https://github.com/wagtail/wagtail/issues/8383
-  Migrates the rest of the reference/contrib section pages from RST to MD
-   [X] Does the code comply with the style guide?
-   [X] For new features: Has the documentation been updated accordingly?

**Minor change**

* Heading levels for `RoutablePageMixin` have changed - the existing heading levels seemed like a mistake; rendering other pages & reversing URLs are now a h3 not a h2.

### CODE REVIEW

File: docs/reference/contrib/settings.md [hunk 1]
```diff
@@ -148,6 +146,8 @@ def view(request):
     ...
 ```
 
+(site_settings)=
```
Comment (Stormheg) [inline docs/reference/contrib/settings.md:149]: Nice one! 

There is a 'site settings' link on https://docs.wagtail.org/en/stable/advanced_topics/multi_site_multi_instance_multi_tenancy.html that now links to this section thanks to this change 👍

File: docs/reference/contrib/typed_table_block.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ The `typed_table_block` module provides a StreamField block type for building ta
 
 ## Installation
 
-Add `"wagtail.contrib.typed_table_block"` to your INSTALLED_APPS:
+Add `"wagtail.contrib.typed_table_block"` to your `INSTALLED_APPS`:
```
Comment (Stormheg) [inline docs/reference/contrib/typed_table_block.md:7]: Thanks for the fix!

File: docs/reference/contrib/typed_table_block.md [hunk 3]
```diff
@@ -36,20 +36,20 @@ class DemoStreamBlock(blocks.StreamBlock):
...
+    ('image', ImageChooserBlock()),
+    ('country', ChoiceBlock(choices=[
+        ('be', 'Belgium'),
+        ('fr', 'France'),
+        ('de', 'Germany'),
+        ('nl', 'Netherlands'),
+        ('pl', 'Poland'),
+        ('uk', 'United Kingdom'),
+    ])),
+])
```
Comment (Stormheg) [inline docs/reference/contrib/typed_table_block.md:52]: 👍

**Review by Stormheg — APPROVED** [review]
Much appreciated @lb- 👍

Converting all the documentation to markdown is a tedious task. With this PR merged, it looks like everything has been converted to markdown. Except for some old release notes.

Thank you for fixing some of the small issues too 😄

**Comment by lb-** [comment]
Approved - merging in.

File: docs/reference/contrib/settings.md (hunk 1 above)
Comment (lb-) [reply docs/reference/contrib/settings.md:149]: yeah - I forgot we have now got generic settings as of 4.0 yay

**Pull request merged**